src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Do not use this. Use [bit-fields](https://docs.rs/bit-fields).
+#![warn(clippy::pedantic, clippy::restriction)]
+#![allow(
```

Comment:
`#!warn(clippy::pedantic, clippy::restriction)` enables a large number of additional lints. In the case of `clippy::restriction` some of these are even contradictory (whether to do `3_i32` or `3i32`) and some are inconsequential for Firecracker (`alloc` over `std`).

With this I am significantly raising the strictness of clippy so this enhances clippy rather than makes it redundant here.